### ISSUE TITLE
Arelia CLI (au new) does not use latest packages in new projects

### ISSUE THREAD

**Issue by EisenbergEffect** [aurelia/cli#468]
@boshinski commented on [Mon Jan 23 2017](https://github.com/aurelia/framework/issues/701)

**I'm submitting a bug report**

* **Library Version:**
AURELIA-CLI 0.24.0

**Please tell us about your environment:**
* **Operating System:**
Windows 7

* **Node Version:**
4.5.0

* **NPM Version:**
3.10.8

* **JSPM OR Webpack AND Version**
NA

* **Browser:**
NA

* **Language:**
TypeScript 2.0

**Current behavior:**
Creating a new project with aurelia-cli ("au new") does not use the latest packages. For example, created a new TypeScript project and then ran "ncu" and it displayed the following:
```
aurelia-bootstrapper             ^1.0.0  →  ^2.0.1
aurelia-tools                    ^0.2.2  →  ^1.0.0
gulp-typescript                 ^2.13.6  →  ^3.1.4
gulp-tslint                      ^5.0.0  →  ^7.0.1
tslint                          ^3.11.0  →  ^4.3.1
typings                          ^1.3.0  →  ^2.1.0
karma                          ^0.13.22  →  ^1.4.0
karma-chrome-launcher            ^1.0.1  →  ^2.0.0
karma-typescript-preprocessor    ^0.2.1  →  ^0.3.1
```
I believe the issue also occurs in new ESNext projects as well.

**Expected/desired behavior:**

* **What is the expected behavior?**
I would expect it to use the latest released versions of the packages.

* **What is the motivation / use case for changing the behavior?**
Avoid the need for the developer to manually upgrade each of these packages when creating a new Aurelia project with the CLI.

---

@AshleyGrant commented on [Tue Jan 24 2017](https://github.com/aurelia/framework/issues/701#issuecomment-274737435)

For the non-aurelia packages listed, this is expected behavior. We would not want to automatically update the package to a new major version without first testing it. That being said, we can likely update all of these dependencies after we have done some testing to make sure everything still works.

A PR for this would be greatly appreciated, but I will try to work on this on my flight today. Thanks @boshinski!

@EisenbergEffect please move this issue to the CLI repo and assign me, if possible. I'll create a PR later today.

**Comment by EisenbergEffect** [comment]
@AshleyGrant Moved.

**Comment by AshleyGrant** [comment]
Thank you. If you have a moment, would you be able to check if `au test --watch` works with the current set of packages? I'm unable to test this from the plane, but with the updated version of `karma`, I'm not able to get files in `src` to be watched, retranspiled, and retested.

**Comment by boshinski** [comment]
I haven't dug into unit testing with karma yet, but a new project created through the Aurelia CLI seems to run the tests OK
```
Starting 'unit'...
24 01 2017 11:08:09.753:WARN [karma]: No captured browser, open http://localhost:9876/

24 01 2017 11:08:09.763:INFO [karma]: Karma v0.13.22 server started at http://localhost:9876/

24 01 2017 11:08:09.776:INFO [launcher]: Starting browser Chrome

24 01 2017 11:08:11.107:INFO [Chrome 55.0.2883 (Windows 7 0.0.0)]: Connected on socket Bv7K28DeZHE6PGNiAAAA with id 77382675

Chrome 55.0.2883 (Windows 7 0.0.0): Executed 1 of 1 SUCCESS (0.005 secs / 0.001 secs)

```

**Comment by s-hoff** [comment]
So far I did a full update of all packages with every project I did, all of them in TypeScript and with unit tests (using aurelia-testing). The only issue I came across is that when building, gulp-typescript gives me a deprecation warning:
```
gulp-typescript: ts(tsProject) has been deprecated - use .pipe(tsProject(reporter)) instead
  As of gulp-typescript 3.0, .pipe(ts(tsProject)) should be written as .pipe(tsProject()).
  More information: http://dev.ivogabe.com/gulp-typescript-3/
```
This is easily fixed by changing the `.pipe(ts(typescriptCompiler))`line to `.pipe(typescriptCompiler())`

Also, as far as I know, au test --watch only watches the `scripts`and `test`directories anyway and we would have to run an `au build --watch`or `au run --watch`in parallel to get the `src` changes?

**Comment by AshleyGrant** [comment]
Yeah, and I think that an `au test --watch` command that doesn't watch for changes in the `src` folder and rebuild/retest isn't very useful :-)

**Comment by con-ssc** [comment]
@s-hoff I just filed a PR for that, I tried that change and it doesn't seem to break anything else ;)

**Comment by YardGnomeNinja** [comment]
I think my experience is related to this issue. When executing `au new` I receive the following and it hangs:

```
Installing Typings
(node:12148) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): typings WARN deprecated 10/25/2016: "registry:dt/jasmine#2.2.0+20160505161446" is deprecated (updated, replaced or removed)

(node:12148) DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.


+-- angular-protractor (global dev)
+-- jasmine (global dev)
+-- node (global dev)
`-- selenium-webdriver (global dev)
```

I am forced to 'ctrl+c' to terminate. It appears as though the project is created properly, however.

**Comment by AshleyGrant** [comment]
@JeroenVinke I was unable to fix this last month. Would you look in to it, please?

**Comment by JeroenVinke** [comment]
@YardGnomeNinja I think you are experiencing [this issue](https://github.com/aurelia/cli/issues/269)

@AshleyGrant and @s-hoff, the testing related issues seem to be related to [this issue](https://github.com/aurelia/cli/issues/370). Can you confirm? This is likely unrelated to any version updates

**Comment by s-hoff** [comment]
@JeroenVinke I can confirm. This is simply how the `au test` command works as of right now and is unrelated to this issue.

**Comment by sbolel** [comment]
I believe this is also related to #494

**Comment by sbolel** [comment]
I created [this PR](https://github.com/sbolel/cli/pull/1) in my fork.

If someone can please review it, I'll finish off the TODOs and create a PR in this repo.

**Comment by JeroenVinke** [comment]
@sbolel Feel free to open a PR to the CLI so we can do a proper review. Thanks for working on this!

**Comment by sbolel** [comment]
@JeroenVinke my pleasure, PR opened: #499

**Comment by JeroenVinke** [comment]
@AStoker think we can close this one? I wonder if we should look into a more sophisticated solution that can update the aurelia packages automatically

**Comment by AStoker** [comment]
I'll leave it up to @EisenbergEffect on whether it should be closed or not. To a point previously made, we would want to test the CLI with the various upgrade releases before making them default (I imagine). For right now, the packages are upgraded, so that's solved. But how future proof it, that's the question. Perhaps we could make a simple script that goes and gets the latest versions of the Aurelia libraries. That script could be used when we do releases (so it will make it easier to update the packages instead of doing it manually), or a user could call the script and upgrade their packages right there in their project... Thoughts?

**Comment by EisenbergEffect** [comment]
I'll close this and create a new issue to track improvements to the CLI build.
